### ISSUE TITLE
Localize Fable client errors 

### ISSUE THREAD

**Issue by Zaid-Ajaj** [Zaid-Ajaj/Fable.Remoting#59]
Errors generated from the Fable client are sent off to the *global* handlers that are defined when the proxy is created. This is not very helpful, because reacting to these errors locally (where it originated) using a global handler is hard 

> for example in Elmish app, you would have to create a subscription for different kinds of errors and propagate a message to the child program where the error originated telling it what happened

Solution: localize the errors to be able to react to them on call-site just like how the dotnet client does it, here is a snippet from the tests
```fs
(*
    calling the function:
    throwError = fun () -> async { 
        return! failwith "Generating custom server error" 
    }
    with error handler: (fun ex routeInfo -> Propagate ex.Message)
*)
testCaseAsync "IServer.throwError using callSafely" <| async {
    let! result = proxy.callSafely <@ fun server -> server.throwError() @> 
    match result with 
    | Ok value -> failwithf "Got value %A where an error was expected" value
    | Result.Error ex -> 
        match ex with 
        | :? Http.ProxyRequestException as reqEx ->
            let statusCode = reqEx.StatusCode 
            Expect.equal HttpStatusCode.InternalServerError statusCode "The status code is 500"
            let! responseText = reqEx.ReadResponseContent() 
            let json = JToken.Parse(responseText) 
            Expect.isFalse (json.["ignored"].ToObject<bool>()) "Error was not ignored"
            Expect.isTrue (json.["handled"].ToObject<bool>()) "Error was handled" 
            Expect.equal (json.["error"].ToString()) "Generating custom server error" "The error was propagated"
        | other -> Expect.isTrue false "Should not happen"
}
```

**Comment by Zaid-Ajaj** [comment]
Or even for a better api:
```fs
(*
    calling the function:
    throwError = fun () -> async { 
        return! failwith "Generating custom server error" 
    }
    with error handler: (fun ex routeInfo -> Propagate ex.Message)
*)
testCaseAsync "IServer.throwError using callSafely" <| async {
    let! result = proxy.callSafely <@ fun server -> server.throwError() @> 
    match result with 
    | Ok value -> failwithf "Got value %A where an error was expected" value
    | Result.Error ex -> 
        match ex with 
        | :? Http.ProxyRequestException as reqEx ->
            Expect.equal HttpStatusCode.InternalServerError reqEx.StatusCode "The status code is 500"
            Expect.isFalse reqEx.Ignored "Error was not ignore"
            Expect.isTrue reqEx.Handled "Error was handled" 
            Expect.equal (reqEx.ParseErrorAs<string>()) "Generating custom server error" "The error was propagated"
        | other -> Expect.isTrue false "Should not happen"
}
```

**Comment by Zaid-Ajaj** [comment]
Fixed in #58, see [error handling](https://zaid-ajaj.github.io/Fable.Remoting/src/error-handling.html)
